library/common/glfw/text_input_plugin.cc

```diff
@@ -189,10 +188,13 @@ void TextInputPlugin::SendStateUpdate(const TextInputModel &model) {
                          std::make_unique<Json::Value>(model.GetState()));
 }
 
-void TextInputPlugin::EnterPressed(const TextInputModel &model) {
+void TextInputPlugin::EnterPressed(TextInputModel &model) {
+  model.AddCharacter('\n');
```

Comment:
If Flutter is appending a newline conditionally, why this this code also unconditionally appending one? Won't that either result in two newlines (for multiline) or an undesired newline (for single-line)?

Reply:
The Flutter engine doesn't add the new line. Seems like it expects the client to add it themselves.

https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/editable_text.dart#L616

Reply:
Ah, I see. Should we really be adding it for single-line fields though? It seems like that would result in the value having a trailing newline on the Dart side. Perhaps we should be checking the input type in setClient (see [here](https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm#L719) for iOS code changing the OS-level input type, for instance), and then keying off that here to decide whether to append the character?

Reply:
Yeah it did seem a bit odd to me. I hadn't seen the info that comes from setClient, that's super helpful. I've fixed the Linux side so far, working on macOS now. Thanks.